Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,624 @@
+
+FileSizeValidator
+=================
+
+The :php-short:`TYPO3\CMS\Extbase\Validation\Validator\FileSizeValidator`
+validates, that the given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects do not exceed the file size configured via the options.
+
+Options:
+
+`minimum`
+    The minimum file size to accept
+`maximum`
+    The maximum file size to accept
+
+Internally the :php:`\TYPO3\CMS\Core\Resource\Security\FileNameValidator` is
+used to validate the file name.
+
+Internally :php:`\TYPO3\CMS\Core\Type\File\FileInfo` is used to determine the
+size.
+
+..  include:: _NoExamples.rst.txt
+
+..  _extbase-validator-float:
+
+FloatValidator
+==============
+
+Checks if a value is a float.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Float'])]
+    protected float $price;
+
+..  _extbase-validator-image-dimensions:
+
+ImageDimensionsValidator
+========================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\ImageDimensionsValidator`
+validate image dimensions of a given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects.
+
+Options:
+
+`width`
+    The exact width of the image
+`height`
+    The exact height of the image
+`minWidth`
+    The minimum width of the image
+`maxWidth`
+    The maximum width of the image
+`minHeight`
+    The minimum height of the image
+`maxHeight`
+    The maximum height of the image
+
+..  include:: _NoExamples.rst.txt
+
+..  _extbase-validator-integer:
+
+IntegerValidator
+================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\IntegerValidator`
+ensures that a value is an integer.
+
+This validator is useful for validating numeric fields that must contain whole
+numbers, such as quantities, IDs, or counters.
+
+#[Validate(['validator' => 'Integer'])]
+protected mixed $quantity;
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Integer'])]
+    protected mixed $quantity;
+
+..  _extbase-validator-mime-type:
+
+MimeTypeValidator
+=================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\MimeTypeValidator`
+validates mime types of a given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects.
+
+Does also validate, if the given file extension matches allowed file extensions
+for the detected mime type.
+
+Options:
+
+`allowedMimeTypes`
+    Allowed mime types (using */* IANA media types)
+
+`ignoreFileExtensionCheck`
+    If set to "true", the file extension check is disabled.
+    Be aware of security considerations when setting this to "true".
+
+..  include:: _NoExamples.rst.txt
+
+..  _extbase-validator-notempty:
+
+NotEmptyValidator
+=================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\NotEmptyValidator`
+ensures that a value is not considered empty.
```

Comment:
Related to this we should note either once and prominently or for each validator that by default validators accept empty values, as in: validation is skipped.

Reply:
done